noir-projects/aztec-nr/aztec/src/encrypted_logs/log_assembly_strategies/default_aes128/note/note.nr

```diff
@@ -0,0 +1,231 @@
+ *                             note_type_id: [u8; 32],
+ *                             ...note: [u8; N * 32],
+ *                         }: [u8; N * 32 + 64],
+ *                     }: [u8; N * 32 + 64],
+
+ *                 ]: [u8; N * 32 + 64],
+ *
+ *                 // the aes128_encrypt fn automatically inserts padding:
+ *                 pt_aes_padding: [u8; 16 - ( (|pt_length| + |pt|) % 16 )]
+ *
+ *             ): [u8; |pt| + |pt_aes_padding|]
+ *                [u8; |ct|]
+ *
+ *             log_bytes_padding_to_mult_31: [u8; 31 * ceil((1 + 48 + |ct|)/31) - (1 + 48 + |ct|)],
+ *                                           [u8; p]
+ *
+ *         ]: [u8; 1 + 48 + |ct| + p]
+ *
+ *     ): [Field; (1 + 48 + |ct| + p) / 31]
+ *
+ *     log_fields_padding: [Field; PRIVATE_LOG_SIZE_IN_FIELDS - 2 - (1 + 48 + |ct| + p) / 31],
+ *
+ * ]: [Field; PRIVATE_LOG_SIZE_IN_FIELDS]
+ *
+ *
+ */
+
+/********************************************************/
+// End of disgusting arithmetic on generics
+/********************************************************/
+
+// TODO: it feels like this existence check is in the wrong place. In fact, why is it needed at all? Under what circumstances have we found a non-existent note being emitted accidentally?
+fn assert_note_exists<let N: u32>(context: PrivateContext, note_hash_counter: u32) {
+    // TODO(#8589): use typesystem to skip this check when not needed
+    let note_exists =
+        context.note_hashes.storage().any(|n: NoteHash| n.counter == note_hash_counter);
+    assert(note_exists, "Can only emit a note log for an existing note.");
+}
+
+/// This particular log assembly strategy (AES 128) requires the note (and the
+/// note_id and the storage_slot) to be converted into bytes, because the aes function
+/// operates on bytes; not fields.
+/// NB: The "2" in "N + 2" is for the note_id and the storage_slot of the note:
+fn compute_note_plaintext_for_this_strategy<Note, let N: u32>(
+    note: Note,
+    storage_slot: Field,
+) -> [Field; (N + 2)]
+where
+    Note: NoteType + Packable<N>,
+{
+    let packed_note = note.pack();
+
+    let mut fields = [0; N + 2];
+    fields[0] = storage_slot;
+    fields[1] = Note::get_id(); // TODO(#10952): The note type ID can be reduced to 7 bits
+    for i in 0..packed_note.len() {
+        fields[i + 2] = packed_note[i];
+    }
+
+    fields
+}
+
+pub fn compute_log<Note, let N: u32>(
+    context: PrivateContext,
+    note: Note,
+    storage_slot: Field,
+    recipient: AztecAddress,
+    sender: AztecAddress,
+) -> [Field; PRIVATE_LOG_SIZE_IN_FIELDS]
+where
+    Note: NoteType + Packable<N>,
+{
+    let final_plaintext = compute_note_plaintext_for_this_strategy(note, storage_slot);
+
+    // AES 128 operates on bytes, not fields, so we need to convert the fields to bytes.
+    // (This process is then reversed when processing the log in `do_process_log`)
+    let final_plaintext_bytes = fields_to_be_bytes_32(final_plaintext);
```

Comment:
This is new here. As for the event I now use this method to have a guraanteed match between encryption and decryption.